src/runtime-rs/crates/resource/src/network/network_with_netns.rs

```diff
@@ -74,7 +74,9 @@ impl NetworkWithNetns {
 impl Network for NetworkWithNetns {
     async fn setup(&self, h: &dyn Hypervisor) -> Result<()> {
         let inner = self.inner.read().await;
-        let _netns_guard = netns::NetnsGuard::new(&inner.netns_path).context("net netns guard")?;
+        if let Some(netns) = &inner.netns_option {
+            netns.enter().context("failed to enter network namespace")?;
```

Comment:
Will `netns` be dropped after exit from `if let Some()` scope? Should the `e.endpoint.attach()` run in the netns?

Reply:
I think it doesn't matter, `netns.enter()` will put the calling thread into the `netns`. Yes, all runtime the threads which do the operation using `netlink` should be set in `netns`

Reply:
I'am agree with @liubin , the `netns` should drop after exit, and his problem occurs many times in patch.

Reply:
The thing is after handling the link, we need to set the runtime threads to back to the previous namespace

Reply:
The thing is we need to set the thread back to previous namespace after handling the link.